### PR TITLE
Manual Edits to `snippets.manual_edits_json` and can be viewed in the files view

### PR DESCRIPTION
**Pull request by RohittCodes**
included manual-edits.json file and implemented update function for it.

image

fixes: #221 
/claim #221

### CODE REVIEW

**Comment by RohittCodes** [comment]
on the right track?? cc: @seveibar

**Comment by RohittCodes** [comment]
@seveibar do i need to implement anything else for the code editor page.

ref: https://github.com/tscircuit/snippets/issues/221#issuecomment-2478498733

File: fake-snippets-api/routes/api/snippets/download.ts [hunk 1]
```diff
@@ -74,6 +74,13 @@ export default withRouteSpec({
         time: snippet.updated_at,
         size: (snippet.dts || "").length,
       },
+      {
+        type: "file",
+        name: "manual-edits.json",
+        hash: "placeholder_hash",
+        time: snippet.updated_at,
+        size: snippet.manual_edits_json?.length,
+      },
```
Comment (seveibar) [inline fake-snippets-api/routes/api/snippets/download.ts:83]: remove this

**Comment by seveibar** [comment]
@RohittCodes the PR isn't complete, as i mentioned in the comment above

**Review by seveibar — APPROVED** [review]
nice work

**Comment by seveibar** [comment]
@RohittCodes CC @imrishabh18 this PR should have a playwright test to prevent this feature from breaking, but I'll create a future issue for it since that policy is new

**Pull request merged**